### PR TITLE
docs(plan): dettaglio feature partner/reseller codes (programma NDS) in v1.4.0

### PR DESCRIPTION
**Pull request by dstmrk**
Aggiunge una sezione autoconsistente a PLAN.md che descrive il modello
wholesale B2B con codici univoci monouso (prefisso per-partner), il modello
dati (tabella partner_codes + colonna referred_by), i flussi di registrazione,
reporting/fatturazione e sospensione, edge case, touchpoint legali/help e la
suddivisione in sub-task per l'implementazione.

https://claude.ai/code/session_01392AosLC7jjMMWUaKLo2Cj

### CODE REVIEW

File: PLAN.md [hunk 1]
```diff
@@ -23,6 +23,214 @@ La versione pubblicata corrente è in `package.json`. Lo storico delle release 
...
+3. In `signUp` (`src/server/auth-actions.ts`): dopo la validazione formato,
+   **claim atomico race-safe** in `insertProfileOrRollback` (o helper
+   dedicato), nella stessa transazione dell'insert profilo:
+   ```
+   UPDATE partner_codes
+      SET used_by = $authUserId, used_at = now()
+    WHERE code = $code AND used_by IS NULL
+   RETURNING code;
+   ```
+   - 0 righe → codice inesistente o già usato → vedi gestione sotto.
```
[the remark this answers is not in the document]
Comment (dstmrk) [reply PLAN.md]: Buona presa, recepito nello spec (commit 52cbbd2). Due modifiche:

1. **Pre-check del codice prima di `supabase.auth.signUp()`** (specularmente al pre-check email già presente): codice mancante/già usato → rifiuto immediato **senza creare l'auth user**, quindi nessun orfano nel caso comune.
2. **Compensating delete esplicito** sul percorso di rifiuto post-creazione: nel caso raro di race TOCTOU (codice consumato tra pre-check e claim atomico, ad auth user già creato), il claim restituisce 0 righe → si invoca la stessa `compensatingDeleteAuthUser` usata per il profile insert failure + errore esplicito. Niente auth user orfano, l'utente può ritentare con la stessa email.

Aggiunti anche i due edge case relativi nella sezione test.

---
_Generated by [Claude Code](https://claude.ai/code)_
